tests/generators/test_nonuniform.py

```diff
@@ -0,0 +1,30 @@
+import xgi
+import pytest
+
+from xgi.exception import XGIError
+
+
+def test_erdos_renyi_hypergraph():
+    with pytest.raises(XGIError):
+        H = xgi.erdos_renyi_hypergraph(10, 20, -0.1)
+    with pytest.raises(XGIError):
+        H = xgi.erdos_renyi_hypergraph(10, 20, 0.0)
+    with pytest.raises(XGIError):
+        H = xgi.erdos_renyi_hypergraph(10, 20, 1.0)
+    with pytest.raises(XGIError):
```

Comment:
I think we should allow for `p` to be 0.0 or 1.0

Reply:
Also, if `p` is not within 0 and 1, we should perhaps raise `ValueError` and not `XGIError`.

Reply:
should be fixed now!